woocommerce/wc-admin
# Hook Reference: Automate a reference resource

Fixes https://github.com/woocommerce/woocommerce-admin/issues/2924

Process all document blocks and compile those that describe hooks into a JSON file for easy data source. This data source can be used to populate a page in the WooCommerce Developer Portal so that extension developers can easily find documentation about available hooks. cc @loranallensmith 

### Next Steps

1. Add `@hook` document blocks to remaining hooks in JavaScript.
2. Figure out how to handle PHP hooks introduced in wc-admin. Should they simply be included in Core's references? Or should they be a part of this process?
2. Write a markdown file describing requirements for developers adding or changing hooks.
3. Create Developer Portal page.

### Detailed test instructions:

1. Run the script
```
npm run create-hook-reference  
```
2. Check out the resulting JSON in `bin/hook-reference/data.json`

```json
[
    {
        "description": "List of homepage stats enabled by default",
        "sourceFile": "https://github.com/woocommerce/woocommerce-admin/blob/1682331e6fe5e898d41049baf41a7da4270a6865/client/homescreen/stats-overview/defaults.js#L5-L16",
        "name": "woocommerce_admin_homepage_default_stats",
        "example": "addFilter( 'woocommerce_admin_homepage_default_stats', 'plugin-domain', ( defaultStats ) => defaultStats.filter( ( stat ) => stat !== 'jetpack/stats/views' ) );"
    }
]
```
3. Make sure the source file url works.
4. Should any other information be added or improved?
